client/rest-high-level/src/test/java/org/elasticsearch/client/indices/GetIndexTemplatesResponseTests.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.indices;
+
+import com.carrotsearch.hppc.cursors.ObjectCursor;
+
+import org.elasticsearch.cluster.metadata.AliasMetaData;
+import org.elasticsearch.cluster.metadata.MappingMetaData;
+import org.elasticsearch.common.bytes.BytesArray;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.xcontent.ToXContent;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.common.xcontent.XContentType;
+import org.elasticsearch.index.mapper.MapperService;
+import org.elasticsearch.test.ESTestCase;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import static org.elasticsearch.test.AbstractXContentTestCase.xContentTester;
+
+public class GetIndexTemplatesResponseTests extends ESTestCase {
+    
+    static final String mappingString = "{\"properties\":{"
+            + "\"f1\": {\"type\":\"text\"},"
+            + "\"f2\": {\"type\":\"keyword\"}"
+            + "}}";
+    
+
+    public void testFromXContent() throws IOException {
```

Comment:
@cbuescher and I had a discussion about how to best test these response classes. We settled on the following approach for `GetIndexResponseTests` and `GetMappingsResponseTests`: https://github.com/elastic/elasticsearch/pull/37796/files/#diff-f66406b858061334c3239c8d2148b2caR43. In short,  we first create a random server object, serialize it to xContent, then parse it back as a client object. We check equality by converting the parsed client object to a server one, and comparing it to the original. This approach lets us avoid duplicating the complex `toXContent` logic in tests.

If you're interested in the full discussion, it's in this comment thread: https://github.com/elastic/elasticsearch/pull/37778/files/cc035c0fa9c505e05169838b8e6e0f8756885662#r250386399

Reply:
I had a very similar thought process. I think I got blocked by the visibility of certain constructors on the server side's POJOs but I can take another look, changing visibility where required if this is the direction we're moving tests in.

Reply:
I added steps 2 and 3 in the latest change to the overall testing process:
1) create a random HLRC response, 
2) Use Java mapping code to copy HLRC attributes to server-side counterpart
3) Call toXContent on the server-side object
4) Use fromXContent to create a new HLRC response
5) Use .equals() call on the HLRC objects from steps 1 )and 4)

Reply:
This looks great! I ended up following the exact same pattern in the 'get mappings' PR. I also caught up with @hub-cap, and confirmed that he is okay with this approach to testing for now.